app/main/views/users.py

```diff
@@ -271,71 +281,123 @@ def export_users_for_framework(framework_slug):
+                Service.framework_id == framework.id
+            ).group_by(
+                Service.supplier_id,
+                Service.lot_id,
+            ).all(),
+            key=lambda row: row[0],
+        )
+    }
 
     supplier_frameworks_and_users = db.session.query(
-        SupplierFramework, User
+        SupplierFramework, Supplier, ContactInformation
     ).filter(
-        SupplierFramework.supplier_id == User.supplier_id
+        SupplierFramework.supplier_id == Supplier.supplier_id
     ).filter(
         SupplierFramework.framework_id == framework.id
     ).filter(
-        User.active.is_(True)
+        ContactInformation.supplier_id == Supplier.supplier_id
     ).options(
-        lazyload(User.supplier),
-        lazyload(SupplierFramework.supplier),
         lazyload(SupplierFramework.framework),
         lazyload(SupplierFramework.prefill_declaration_from_framework),
         lazyload(SupplierFramework.framework_agreements),
     ).order_by(
-        SupplierFramework.supplier_id,
-        User.id,
+        Supplier.supplier_id,
     ).all()
 
-    user_rows = []
-
-    for sf, u in supplier_frameworks_and_users:
-
-        # always get the declaration status
-        declaration_status = sf.declaration.get('status') if sf.declaration else 'unstarted'
-        application_status = 'application' if (
-            declaration_status == 'complete' and sf.supplier_id in suppliers_with_a_complete_service
-        ) else 'no_application'
-        application_result = ''
-        framework_agreement = ''
-        variations_agreed = ''
-
-        # if framework is pending, live, or expired
-        if framework.status != 'open':
-            if sf.on_framework is None:
-                application_result = 'no result'
-            else:
-                application_result = 'pass' if sf.on_framework else 'fail'
-            framework_agreement = bool(getattr(sf.current_framework_agreement, 'signed_agreement_returned_at', None))
-            variations_agreed = ', '.join(sf.agreed_variations.keys()) if sf.agreed_variations else ''
-
-        user_rows.append({
-            'email address': u.email_address,
-            'user_name': u.name,
-            'user_research_opted_in': u.user_research_opted_in,
-            'supplier_id': sf.supplier_id,
-            'declaration_status': declaration_status,
-            'application_status': application_status,
-            'framework_agreement': framework_agreement,
-            'application_result': application_result,
-            'variations_agreed': variations_agreed,
-            'published_service_count': supplier_id_published_service_count.get(sf.supplier_id, 0)
-        })
-
-    return jsonify(users=[user for user in user_rows]), 200
+    supplier_rows = []
+
+    for sf, supplier, ci in supplier_frameworks_and_users:
+        # Only export suppliers that have users
+        if users_for_each_supplier.get(supplier.supplier_id):
```

Comment:
This would be a good case to use a `continue` statement. It will allow you to avoid indenting all the code after the if and it also works with the narrative. 
`if users_for_each_supplier.get(supplier.supplier_id)` is designed to stop us from exporting suppliers with no users so `if not:` `continue` works quite well.

`continue` statements are best used sparingly because when they are hidden in a big block of code they can obscure the flow but their use as a preconditional check at the top of a block is more obvious.